### PR TITLE
Modify slow running HMC unit tests to run faster

### PR DESCRIPTION
**Pull request by neerajprad**
- Removed gmm test, since we have the same one in `test_nuts.py` using pretty much the same code path (i.e. einsum evaluator and potential energy computation).
 - ~Specified an initial step size for `test_gamma_normal`~ (Edit - turned off step size adaptation) which was going really slow with a very small step size (the initialization seems to have changed upon merge of pytoch-1.0 branch).

### CODE REVIEW

**Comment by neerajprad** [comment]
I may modify the tests further if any of them take longer than 60s.

**Comment by fehiepsi** [comment]
@neerajprad I think that you can remove all tests with trailing `dual_averaging` too. We enable adapt_step_size by default, so there is no need for these tests.

File: tests/infer/mcmc/test_hmc.py [hunk 1]
```diff
@@ -298,39 +298,13 @@ def model(data):
 
     true_std = torch.tensor([0.5, 2])
     data = dist.Normal(3, true_std).sample(sample_shape=(torch.Size((2000,))))
-    hmc_kernel = HMC(model, trajectory_length=1, adapt_step_size=True,
+    hmc_kernel = HMC(model, trajectory_length=1, step_size=0.3,
```
[the remark this answers is not in the document]
Comment (fehiepsi) [reply tests/infer/mcmc/test_hmc.py]: I suggest to set this to a smaller step_size ((because things are wrong with even very small step_size, I suspect that initial trace is so bad; so we should use a small step_size for mass_adaptation; 0.3 is too large for trajectory_length=1 and bad initial trace), and set `adapt_step_size=False`. You might increase `warmup_steps` to a higher value if needed.

[the remark this answers is not in the document]
Comment (neerajprad) [reply tests/infer/mcmc/test_hmc.py]: I'll just change the rng seed and increase the number of warmup steps. Otherwise we'll need a very small step size without adaptation, and that will take even longer to finish.

[the remark this answers is not in the document]
Comment (fehiepsi) [reply tests/infer/mcmc/test_hmc.py]: How about `step_size=0.03, adapt_step_size=False` and `warmup_steps=200`? It works in my machine. I don't see the reason for changing init_step_size and still use adapt_step_size.

[the remark this answers is not in the document]
Comment (fehiepsi) [reply tests/infer/mcmc/test_hmc.py]: Ah, by `init_step_size`, I mean the input `step_size`. Sorry for making confusion!

[the remark this answers is not in the document]
Comment (neerajprad) [reply tests/infer/mcmc/test_hmc.py]: Np, I meant step size too! I had changed the random seed without specifying the `step_size`, but this is probably going to be more stable. Relying on fixing the seed isn't a stable strategy specially with the ongoing refactoring of the rng in pytorch.
